modules/core/04-channel/types/packet.go

```diff
@@ -35,6 +55,51 @@ func CommitPacket(cdc codec.BinaryCodec, packet Packet) []byte {
 	return hash[:]
 }
 
+// CommitEurekaPacket returns the Eureka packet commitment bytes. The commitment consists of:
+// sha256_hash(timeout_timestamp + timeout_height.RevisionNumber + timeout_height.RevisionHeight + sha256_hash(data))
+// from a given packet. This results in a fixed length preimage.
+// NOTE: A fixed length preimage is ESSENTIAL to prevent relayers from being able
+// to malleate the packet fields and create a commitment hash that matches the original packet.
+// NOTE: sdk.Uint64ToBigEndian sets the uint64 to a slice of length 8.
+func CommitEurekaPacket(packet Packet) []byte {
+	timeoutHeight := packet.GetTimeoutHeight()
+
+	timeoutBuf := sdk.Uint64ToBigEndian(packet.GetTimeoutTimestamp())
+
+	// only hash the timeout height if it is non-zero. This will allow us to remove it entirely in the future
+	if !reflect.DeepEqual(timeoutHeight, clienttypes.ZeroHeight()) {
+		revisionNumber := sdk.Uint64ToBigEndian(timeoutHeight.GetRevisionNumber())
+		timeoutBuf = append(timeoutBuf, revisionNumber...)
+
+		revisionHeight := sdk.Uint64ToBigEndian(timeoutHeight.GetRevisionHeight())
+		timeoutBuf = append(timeoutBuf, revisionHeight...)
+	}
+
+	// hash the timeout rather than using a fixed-size preimage directly
+	// this will allow more flexibility in the future with how timeouts are defined
+	timeoutHash := sha256.Sum256(timeoutBuf)
+	buf := timeoutHash[:]
+
+	// hash the destination identifiers since we can no longer retrieve them from the channelEnd
+	portHash := sha256.Sum256([]byte(packet.GetDestPort()))
+	buf = append(buf, portHash[:]...)
+	destinationHash := sha256.Sum256([]byte(packet.GetDestChannel()))
+	buf = append(buf, destinationHash[:]...)
+
+	// hash the version only if it is nonempty
```

Comment:
why? What's wrong with hashing the version even if it is empty

Reply:
We could do this, thought to save on the compute if hash is empty. More relevant for VM implementations

Reply:
I think it is better not to have branching in the code when not necessary. A single hash isn't gonna make or break it. So can we hash even if it is empty?